### PR TITLE
doc/devguide: add transactions guide for devs

### PR DESCRIPTION
**Pull request by jufajardini**
Describe changes:
- incorporate grammar corrections
- update DNS sequence diagram
- change location of script for image generation
- squash commits

Link to Redmine issue: https://redmine.openinfosecfoundation.org/issues/4396

Previous PR: #6268 

Current state with images looks like this: https://devguide-transactions.surge.sh/extending/app-layer/transactions.html

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings.
Pipeline 3624

**Comment by suricata-qa** [comment]
Information: QA ran without warnings.

Pipeline 3624

**Comment by suricata-qa** [comment]
Information: QA ran without warnings.

Pipeline 3624

**Comment by suricata-qa** [comment]
Information: QA ran without warnings.

Pipeline 3624

**Comment by suricata-qa** [comment]
Information: QA ran without warnings.

Pipeline 3624

**Comment by suricata-qa** [comment]
Information: QA ran without warnings.

Pipeline 3624

**Comment by suricata-qa** [comment]
Information: QA ran without warnings.

Pipeline 3624

**Comment by suricata-qa** [comment]
Information: QA ran without warnings.

Pipeline 3624

File: doc/devguide/extending/app-layer/transactions.rst [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+.. code-block:: c
+
+    AppLayerParserRegisterStateProgressCompletionStatus(
+        ALPROTO_FTP, FTP_STATE_FINISHED, FTP_STATE_FINISHED);
+
+Sequence Diagrams
+~~~~~~~~~~~~~~~~~
+
+A DNS transaction in Suricata can be considered unidirectional:
```
Comment (victorjulien) [inline doc/devguide/extending/app-layer/transactions.rst:217]: The image here shows that both request and response are part of the transaction, which would make it bidirectional. The implementation for dns is unidirectional, which means that both request and response get their own tx.

File: doc/devguide/extending/app-layer/transactions.rst [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+
+.. image:: img/HTTP2BidirectionalTransaction.png
+  :width: 600
+  :alt: A sequence diagram with two entities, Client and Server, with an arrow going from the Client to the Server labeled "Request" and below that an arrow going from Server to Client labeled "Response". Below those arrows, a dotted line indicates that the transaction is completed.
+
+A TLS Handshake is a more complex example, where several messages are exchanged before the transaction is considered completed:
+
+.. image:: img/TlsHandshake.png
+  :width: 600
+  :alt: A sequence diagram with two entities, Client and Server, with an arrow going from the Client to the Server labeled "ClientHello" and below that an arrow going from Server to Client labeled "ServerHello". Below those arrows, several more follow from Server to Client and vice-versa, before a dotted line indicates that the transaction is finally completed.
```
Comment (victorjulien) [inline doc/devguide/extending/app-layer/transactions.rst:236]: The actual progress used in the protocol parser implementation is a lot simpler:
```c
enum {
    TLS_STATE_IN_PROGRESS = 0,
    TLS_STATE_CERT_READY = 1,
    TLS_HANDSHAKE_DONE = 2,
    TLS_STATE_FINISHED = 3
};
```
I think it makes sense to map the protocol states to the simplified protocol parser states?

File: doc/devguide/extending/app-layer/transactions.rst (hunk 1 above)
Comment (jufajardini) [reply doc/devguide/extending/app-layer/transactions.rst:217]: Hm, I updated this following a comment from @jasonish in my previous pr (https://github.com/OISF/suricata/pull/6268#discussion_r675693271)
I probably got it wrong and the idea was to have another unidirectional transaction, but representing just the response?

**Comment by jufajardini** [comment]
Followed by: https://github.com/OISF/suricata/pull/6321

**Pull request closed without merging**